lib/battle/CBattleInfoEssentials.h

```diff
@@ -0,0 +1,108 @@
+/*
+ * CBattleInfoEssentials.h, part of VCMI engine
+ *
+ * Authors: listed in file AUTHORS in main folder
+ *
+ * License: GNU General Public License v2.0 or later
+ * Full text of license available in license.txt file, in main folder
+ *
+ */
+#pragma once
+#include "CCallbackBase.h"
+
+class CGTownInstance;
+class CGHeroInstance;
+class CStack;
+struct CObstacleInstance;
+class IBonusBearer;
+struct InfoAboutHero;
+class CArmedInstance;
+
+typedef std::vector<const CStack*> TStacks;
+typedef std::function<bool(const CStack *)> TStackFilter;
+
+namespace BattlePerspective
+{
+enum BattlePerspective
+{
+	INVALID = -2,
+	ALL_KNOWING = -1,
+	LEFT_SIDE,
+	RIGHT_SIDE
+};
+}
+
+namespace BattleSide
+{
+enum {ATTACKER = 0, DEFENDER = 1};
+}

```

Comment:
Indent namespaces with tabs. Some probably could be replaced with strongly typed enum.